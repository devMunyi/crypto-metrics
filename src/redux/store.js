import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import cryptoReducer from './reducers';

export default configureStore(
  {
    reducer: {
      crypto: cryptoReducer,
    },
  },
  applyMiddleware(logger),
);
