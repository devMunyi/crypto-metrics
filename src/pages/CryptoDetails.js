import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { SyncOutlined } from '@ant-design/icons';
import { BsFillArrowRightCircleFill } from 'react-icons/bs';
import { fetchSingleCrypto } from '../redux/actions';
import Headline from '../components/Headline';
/*eslint-disable */

function CryptoDetails() {
  // get crypto id from dynamic route params
  const { cryptoId } = useParams();

  // set dispatch hook
  const dispatch = useDispatch();

  // fetch a single crypto by id on pgae mount
  useEffect(() => {
    dispatch(fetchSingleCrypto(cryptoId));
  }, [cryptoId, dispatch]);

  // get single crypto details from the store
  const { cryptoDetails, isLoading } = useSelector((state) => state.crypto);

  return (
    <div className="container text-white">
      {isLoading && (
        <SyncOutlined
          spin
          style={{ fontSize: '32px' }}
          className="d-flex justify-content-center align-items-center"
        />
      )}
      {!isLoading && (
        <>
          <Headline
            page="details"
            cryptoDetails={cryptoDetails}
            cryptoCount={0}
          />

          <div className="row pt-3 pb-3 container fw-normal">
            PRIMARY INFORMATION:
          </div>

          <table className="table table-striped">
            <tbody>
              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  Name
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.name}
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>
              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  Rank
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.rank}
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>
              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  Price
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.priceUsd} usd
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>

              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  changePercent24Hr
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.changePercent24Hr} usd
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>

              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  Symbol
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.symbol}
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>

              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  MarketCapUsd
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.marketCapUsd}
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>

              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  Supply
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.supply}
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>
              <tr>
                <td style={{ width: '60%', border: '2px solid white' }}>
                  MaxSupply
                </td>
                <td style={{ width: '20%', border: '2px solid white' }}>
                  {cryptoDetails.maxSupply ? cryptoDetails.maxSupply : 'N/A'}
                </td>
                <td style={{ width: '10%', border: '2px solid white' }}>
                  <a style={{ color: '#FFF' }} href={cryptoDetails.explorer}>
                    <BsFillArrowRightCircleFill />
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </>
      )}
    </div>
  );
}

export default CryptoDetails;
