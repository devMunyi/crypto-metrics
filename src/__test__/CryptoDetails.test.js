import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import CryptoDetails from '../pages/CryptoDetails';

/* eslint-disable comma-dangle */
it('renders CryptoDetails Component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <CryptoDetails />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
