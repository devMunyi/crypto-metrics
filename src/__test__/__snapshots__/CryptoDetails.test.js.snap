// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CryptoDetails Component correctly 1`] = `
<div
  className="container text-white"
>
  <div
    className="row border headline"
    style={
      Object {
        "boxShadow": "rgba(0, 0, 0, 0.24) 0px 3px 8px",
      }
    }
  >
    <div
      className="col"
      style={
        Object {
          "height": "200px",
        }
      }
    >
      <img
        alt="CRYPTO"
        className="img-fluid"
        src="cryptos.png"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="col d-flex justify-content-center align-items-center border-start"
    >
      <div
        className="div"
      >
        <h4 />
        <p
          className="fw-normal"
        >
           
          undefined usd
        </p>
      </div>
    </div>
  </div>
  <div
    className="row pt-3 pb-3 container fw-normal"
  >
    PRIMARY INFORMATION:
  </div>
  <table
    className="table table-striped"
  >
    <tbody>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          Name
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        />
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          Rank
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        />
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          Price
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        >
           usd
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          changePercent24Hr
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        >
           usd
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          Symbol
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        />
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          MarketCapUsd
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        />
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          Supply
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        />
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
      <tr>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "60%",
            }
          }
        >
          MaxSupply
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "20%",
            }
          }
        >
          N/A
        </td>
        <td
          style={
            Object {
              "border": "2px solid white",
              "width": "10%",
            }
          }
        >
          <a
            style={
              Object {
                "color": "#FFF",
              }
            }
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
              />
            </svg>
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
